Upstream-Status: Pending [https://github.com/jenswi-linaro/optee_os/commit/7c3b052b871420618a08402f626923fe85a139db]
Signed-off-by: Arunachalam Ganapathy <arunachalam.ganapathy@arm.com>

From 7c3b052b871420618a08402f626923fe85a139db Mon Sep 17 00:00:00 2001
From: Marc Bonnici <marc.bonnici@arm.com>
Date: Wed, 23 Sep 2020 13:55:22 +0100
Subject: [PATCH 1/3] core: SPMC: update for FF-A version 1.0

Update expected memory attributes. As per the FF-A spec (5.11.3)  a
lender of a memory region should not specify the instruction access
permission, therefore update the expected memory attribute.

Updates the version number to 1.0 replacing the previous version number
0.9.
---
 core/arch/arm/include/optee_ffa.h  | 4 ++--
 core/arch/arm/kernel/thread_spmc.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/arch/arm/include/optee_ffa.h b/core/arch/arm/include/optee_ffa.h
index 8498fbab..9daca732 100644
--- a/core/arch/arm/include/optee_ffa.h
+++ b/core/arch/arm/include/optee_ffa.h
@@ -27,8 +27,8 @@
  * w3-w7: Implementation defined, free to be used below
  */
 
-#define OPTEE_FFA_VERSION_MAJOR	UINT32_C(0)
-#define OPTEE_FFA_VERSION_MINOR	UINT32_C(9)
+#define OPTEE_FFA_VERSION_MAJOR	UINT32_C(1)
+#define OPTEE_FFA_VERSION_MINOR	UINT32_C(0)
 
 #define OPTEE_FFA_BLOCKING_CALL(id)	UINT32_C(id)
 #define OPTEE_FFA_YIELDING_CALL_BIT	31
diff --git a/core/arch/arm/kernel/thread_spmc.c b/core/arch/arm/kernel/thread_spmc.c
index 81edf03a..16f3bbe1 100644
--- a/core/arch/arm/kernel/thread_spmc.c
+++ b/core/arch/arm/kernel/thread_spmc.c
@@ -433,7 +433,7 @@ static int mem_share_init(void *buf, size_t blen, unsigned int *page_count,
 {
 	struct mem_region_descr *region_descr = NULL;
 	struct mem_transaction_descr *descr = NULL;
-	const uint8_t exp_mem_acc_perm = 0x6; /* Not executable, Read-write */
+	const uint8_t exp_mem_acc_perm = 0x2; /* Read-write */
 	/* Normal memory, Write-Back cacheable, Inner shareable */
 	const uint8_t exp_mem_reg_attr = 0x2f;
 	unsigned int num_mem_accs = 0;
-- 
2.26.2

